css-scroll-snap-2/Overview.bs

```diff
@@ -275,17 +275,125 @@ Snap Events {#snap-events}
+	{{snapchanged}} should be dispatched:
+
+	<ol>
+	<li>
+		at the end of a scrolling operation, after the user agent has completed its
+		snapping, if either:
+		<ul>
+		<li>	the snap area which the snap container is snapped to (TODO: link spec on how the target is picked) in the block axis is different
+			from the snap area it most recently snapped to in the block axis, or
+		</li>
+		<li>	the snap area it is snapped to in the inline axis is different
+			from the snap area it most recently snapped to in the inline axis.
+		</li>
+		</ul>
+	</li>
+	<li> (for snap containers with ''scroll-snap-type/proximity'' strictness) at the end of a scroll in which
+		the snap container scrolls and settles into (or out of) a
+		snap area's range of proximity snap positions in either axis.
+	</li>
+	<li> if there is a change to a snap container's style such that it goes from having a
+		non-'none' value for scroll-snap-type to having a 'none' value or vice versa.
+	</li>
+	<li> if, after a <a href="https://drafts.csswg.org/css-scroll-snap-1/#re-snap">layout change</a>, the snap area to which a snap container is
+		snapped to changes, regardless of whether there is a change in scroll position
+		after the layout change.
+	</li>
+	</ol>
+
+	Although, snapping occurs when a snap container is first
+	laid out, {{snapchanged}} should not be triggered by this initial layout snap.
+	Scrolling operations always lead to {{scrollend}} events being fired. If a scrolling operation
+	also results in a {{snapchanged}} event being fired, the {{snapchanged}} event should be fired
+	before the {{scrollend}} event.
+
+<h4 id="snapchanging"> snapchanging </h4>
+	{{snapchanging}} is dispatched when the snap area to which a
+	snap container would snap (in either axis) during a scrolling operation changes.
+	A scrolling operation may animate towards a particular position (e.g.
+	scrollbar arrow clicks, arrow key presses, "behavior: smooth" programmatic scrolls)
+	or may directly track a user's input (e.g. touch scrolling, scrollbar dragging).
+	{{snapchanging}} aims to let the web page know, as early as possible,
+	that the scrolling operation will result in a change in the snap area
+	the snap container is snapped to.
+	* In the former case, the user agent should evaluate whether snapchanging is occurring based on
+		the snap position that will eventually be snapped to after the scroll animation's
+		target offset is reached.
+	* In the latter case, the user agent should evaluate whether snapchanging should
+		be triggered based on the snap position that would be snapped to if
+		the scroll is ended at the current scroll position (determined by the user's input).
+
+	Note that since snapchanging gives the web page hints about future snapping,
+	the snapping hinted at by a snapchanging event may not materialize since it will be
+	possible for subsequent scrolling input to further alter the snap container's
+	scroll position and result in a different eventual snap position.
```

Comment:
I suspect it may be clearer to explain the semantics and timing of snapchanging and snapchanged together as an algorithm. E.g. I think we expect that snapchanging is fired before snapchanged, but also that both will be fired in some cases, e.g. if a scroll operation ends unexpectedly early we might fire snapchanging and snapchanged immediately for the current snap location.

Reply:
I've added Appendix B which indicates that snapchanging will fire during scrolling and snapchanged will fire after scrolling is completed.